libs/common/src/dtos/announcement.dto.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * File name should always end with `.dto.ts` for swagger metadata generator to get picked up
+ */
+// eslint-disable-next-line max-classes-per-file
+import { NoteActivityDto, ProfileActivityDto } from './activity.dto';
+
+// eslint-disable-next-line no-shadow
+export enum AnnouncementTypeDto {
+  TOMBSTONE = 'tombstone',
+  BROADCAST = 'broadcast',
+  REPLY = 'reply',
+  REACTION = 'reaction',
+  PROFILE = 'profile',
+  UPDATE = 'update',
+}
+
+export class BroadcastDto extends NoteActivityDto {}
```

Comment:
Is it worth nesting the Activity Content in a property in case there are collisions between the announcement fields and the Activity Content fields? Or does that cause issues with the db setup?

```suggestion
export class BroadcastDto {
  content: NoteActivityDto // Perhaps other key name?
}
```

Or perhaps a parent class? I don't think I like this one as much without generics:
```suggestion
export class BroadcastAnnouncementDto {
  announcement: BroadcastDto,
  content: NoteActivityDto // Perhaps other key name?
}
```

Reply:
So a couple of things for this. NoteActivityDto doesn't have to match the Note Activity Content fields by feild since it's only a subset and DTO for that, so in case of any activity content changes it would not directly effect the dto to cause conflict.

In any case I liked the first one better which might allow for announcement specific fields in case we ned to add any.

Side node: We don't have to make these too much future proof since we can always create new endpoints and dtos with new version which would be compatible with new DSNP spec.